src/components/Select/Select.js

```diff
@@ -0,0 +1,196 @@
+
+  const onChange = (newValue, actionMeta) => {
+    const { action, option, removedValue } = actionMeta;
+
+    switch (action) {
+      case "select-option": {
+        if (option.value === allOption.value) {
+          return initialOnChange(options, actionMeta);
+        }
+        return initialOnChange(newValue || [], actionMeta);
+      }
+      case "deselect-option": {
+        if (option.value === allOption.value) {
+          return initialOnChange([], actionMeta);
+        }
+        if (isAllOptionSelected) {
+          return initialOnChange(
+            options.filter(({ value }) => value !== option.value),
+            actionMeta
+          );
+        }
+        return initialOnChange(newValue || [], actionMeta);
+      }
+      case "remove-value": {
+        if (removedValue.value === allOption.value) {
+          return initialOnChange([], actionMeta);
+        }
+        if (isAllOptionSelected) {
+          return initialOnChange(
+            options.filter(({ value }) => value !== removedValue.value),
+            actionMeta
+          );
+        }
+        return initialOnChange(newValue || [], actionMeta);
+      }
+      default:
+        return initialOnChange(newValue || [], actionMeta);
+    }
+  };
+
+  return {
+    onChange,
+    options: fullOptions,
+    value: isAllOptionSelected ? [allOption, ...value] : value,
+    components: {
+      ...originalProps.components,
+      ValueContainer: SelectAllFeatureValueContainer
+    }
+  };
+};
+
+const selectColourStyles = {
+  control: (styles, { isFocused }) => {
+
+    return {
+      ...styles,
+      minHeight: "42px",
+      border: "1px solid #D1D5DB",
+      borderColor: isFocused ? "#6366F1" : undefined,
+      boxShadow: isFocused
+        ? "0px 0px 0px 2px rgba(99, 102, 241, 1), 0px 1px 2px 0px rgba(0, 0, 0, 0.05)"
+        : "0px 1px 2px rgba(0, 0, 0, 0.05)",
+      borderRadius: "6px",
+      "&:hover": {
+        ...styles["&:hover"],
+        borderColor: undefined,
+        boxShadow: undefined
+      }
+    };
+  },
+  option: (styles, { isSelected, isFocused }) => ({
+    ...styles,
+    backgroundColor: isSelected ? "#4e45df" : isFocused ? "#e6e4fa" : undefined,
+    ":active": {
+      ...styles[":active"],
+      backgroundColor: "#ccc9f6"
+    }
+  }),
+  indicatorSeparator: (styles) => ({
+    ...styles,
+    backgroundColor: "#9CA3AF"
+  }),
+  dropdownIndicator: (styles) => ({
+    ...styles,
+    color: "#9CA3AF"
+  }),
+  clearIndicator: (styles) => ({
+    ...styles,
+    color: "#9CA3AF"
+  }),
+  menu: (styles) => ({
+    ...styles,
+    borderRadius: "6px",
+    boxShadow:
+      "0px 10px 15px -3px rgba(0, 0, 0, 0.1), 0px 4px 6px -2px rgba(0, 0, 0, 0.05), 0px 0px 0px 1px rgba(0, 0, 0, 0.05)"
+  }),
+  singleValue: (styles) => ({
+    ...styles,
+    fontSize: "16px",
+    lineHeight: "24px"
+  })
+};
+
+export const Select = ({ name, control, ...props }) => {
+  return (
+    <Controller
```

Comment:
The `control` is optional you can specify either control or onSelect

Reply:
What's the point in additional onSelect props? You can use original onChange prop... Or you want it to transform somehow value.

Reply:
either onSelect or onChange is fine - by control must be optional

Reply:
done, removed border in the input